packages/repository-modernizer/src/restructure-filters.js

```diff
@@ -64,6 +64,7 @@ var RestructureFilterPaths = {
                         ? project.relativePathToExistingFilterXml
                         : constants.FILTER_PATH
                 );
+                if(!project.relativePathToExistingFilterXml.endsWith("pom.xml")){
```

Comment:
You will also need to check `project.relativePathToExistingFilterXml` for null, or code will throw NPE in case the path is not mentioned.